packages/headless/src/controllers/commerce/search-box/headless-search.box.ts

```diff
@@ -0,0 +1,114 @@
+import {fetchQuerySuggestions} from '../../../features/query-suggest/query-suggest-actions';
+import {logSearchboxSubmit} from '../../../features/query/query-analytics-actions';
+import {executeSearch} from '../../../features/search/search-actions';
+import {SuggestionHighlightingOptions, Delimiters} from '../../../utils/highlight';
+import {Controller} from '../../controller/headless-controller';
+import {
+  buildCoreSearchBox,
+  SearchBoxState,
+  Suggestion,
+} from '../../core/search-box/headless-core-search-box';
+import {CommerceEngine} from '../../../app/commerce-engine/commerce-engine';
+
+export type {
+  SearchBoxState,
+  SuggestionHighlightingOptions,
+  Suggestion,
+  Delimiters,
+};
+
+export interface SearchBoxOptions {
+  /**
+   * A unique identifier for the controller.
+   * By default, a unique random identifier is generated.
+   */
+  id?: string;
+
+  /**
+   * Specifies delimiters to highlight parts of a query suggestion that e.g match, do not match the query.
+   */
+  highlightOptions?: SuggestionHighlightingOptions;
+
+  // TODO(nico): Figure out what to do with this
+  /**
+   * Whether to clear all active query filters when the end user submits a new query from the search box.
+   * Setting this option to "false" is not recommended & can lead to an increasing number of queries returning no results.
+   */
+  clearFilters?: boolean;
+}
+
+export interface SearchBoxProps {
+  /**
+   * The `SearchBox` controller options.
+   */
+  options?: SearchBoxOptions;
+}
+
+/**
+ * The `SearchBox` headless controller offers a high-level interface for designing a common search box UI controller
+ * with [highlighting for query suggestions](https://docs.coveo.com/en/headless/latest/usage/highlighting/).
+ */
+export interface SearchBox extends Controller {
+  /**
+   * Updates the search box text value and shows the suggestions for that value.
+   *
+   * @param value - The string value to update the search box with.
+   */
+  updateText(value: string): void;
+
+  /**
+   * Clears the search box text and the suggestions.
+   */
+  clear(): void;
+
+  /**
+   * Shows the suggestions for the current search box value.
+   */
+  showSuggestions(): void;
+
+  /**
+   * Selects a suggestion and calls `submit`.
+   *
+   * @param value - The string value of the suggestion to select
+   */
+  selectSuggestion(value: string): void;
+
+  /**
+   * Deselects all facets and triggers a search query.
+   */
+  submit(): void;
+
+  /**
+   * The state of the `SearchBox` controller.
+   */
+  state: SearchBoxState;
+}
+
+/**
+ * Creates a `SearchBox` controller instance.
+ *
+ * @param engine - The headless engine.
+ * @param props - The configurable `SearchBox` properties.
+ * @returns A `SearchBox` controller instance.
+ */
+export function buildSearchBox(
+  engine: CommerceEngine,
+  props: SearchBoxProps = {}
+): SearchBox {
+  const searchBox = buildCoreSearchBox(engine, {
+    ...props,
+    executeSearchActionCreator: executeSearch,
```

Comment:
Use a noop here 🤠

Reply:
@alexprudhomme you checked this bit a bit earlier today, can you check this out to make sure we ain't gonna cripple this lil searchbox.

Reply:
Eventually, commerce will send their own executeSearch function so this is fine as a draft for now. Do I understand correctly ?

Reply:
See https://github.com/coveo/ui-kit/pull/3461#discussion_r1421073364